core/src/main/java/hudson/slaves/JNLPLauncher.java

```diff
@@ -195,6 +198,59 @@ public void launch(SlaveComputer computer, TaskListener listener) {
     @Restricted(NoExternalUse.class)
     public static /*almost final*/ Descriptor<ComputerLauncher> DESCRIPTOR;
 
+    @NonNull
+    @Restricted(NoExternalUse.class)
+    public String getRemotingOptionsUnix(@NonNull Computer computer) {
+        return getRemotingOptions(escapeUnix(computer.getName()));
+    }
+
+    @NonNull
+    @Restricted(NoExternalUse.class)
+    public String getRemotingOptionsWindows(@NonNull Computer computer) {
+        return getRemotingOptions(escapeWindows(computer.getName()));
+    }
+
+    private String getRemotingOptions(String computerName) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("-name ");
+        sb.append(computerName);
+        sb.append(' ');
+        if (isWebSocket()) {
```

Comment:
We could simply deprecate this option, since it will now have no effect other than changing the recommendation for the command. Compare discussion of `vmargs` in #6543.

Reply:
So without this change (including upstream), agents automatically apply configuration changes in Jenkins on reconnection (as the JNLP file has these), while now the agent command line needs to changed? That seems confusing/unexpected as an admin and worth documenting as least IMO (and arguably that was a pretty nice feature to have).